jupyter_notebooks/Book_4_Mountain_Wave_Moving_Mesh.ipynb

```diff
@@ -0,0 +1,508 @@
+   "id": "46852826",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "    Residual norms for exner_solver_ solve.\n",
+      "    0 KSP preconditioned resid norm 3.315484606629e+02 true resid norm 4.100966222339e+06 ||r(i)||/||b|| 1.000000000000e+00\n",
+      "    1 KSP preconditioned resid norm 3.465899943748e-10 true resid norm 1.118900482277e-07 ||r(i)||/||b|| 2.728382585018e-14\n",
+      "    Residual norms for exner_solver_ solve.\n",
+      "    0 KSP preconditioned resid norm 3.310172303592e+02 true resid norm 4.091155332443e+06 ||r(i)||/||b|| 1.000000000000e+00\n",
+      "    1 KSP preconditioned resid norm 3.468406051930e-10 true resid norm 8.409519105751e-08 ||r(i)||/||b|| 2.055536498227e-14\n",
+      "    Residual norms for rhosolver_ solve.\n",
+      "    0 KSP preconditioned resid norm 8.093018696236e-05 true resid norm 7.079661084073e+00 ||r(i)||/||b|| 1.000000000000e+00\n",
+      "    1 KSP preconditioned resid norm 2.432996581760e-17 true resid norm 8.572755689479e-15 ||r(i)||/||b|| 1.210899164194e-15\n"
+     ]
+    }
+   ],
+   "source": [
+    "def minimum(f):\n",
+    "    fmin = op2.Global(1, [1000], dtype=float)\n",
+    "    op2.par_loop(op2.Kernel(\"\"\"\n",
+    "static void minify(double *a, double *b) {\n",
+    "    a[0] = a[0] > fabs(b[0]) ? fabs(b[0]) : a[0];\n",
+    "}\n",
+    "\"\"\", \"minify\"), f.dof_dset.set, fmin(op2.MIN), f.dat(op2.READ))\n",
+    "    return fmin.data[0]\n",
+    "\n",
+    "\n",
+    "p0 = minimum(exner)\n",
+    "compressible_hydrostatic_balance(state, theta_b, rho_b, exner,\n",
+    "                                 top=True, params=exner_params)\n",
+    "p1 = minimum(exner)\n",
+    "alpha = 2.*(p1-p0)\n",
+    "beta = p1-alpha\n",
+    "exner_top = (1.-beta)/alpha\n",
+    "compressible_hydrostatic_balance(state, theta_b, rho_b, exner,\n",
+    "                                 top=True, exner_boundary=exner_top, solve_for_rho=True,\n",
+    "                                 params=exner_params)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2b518722",
+   "metadata": {},
+   "source": [
+    "Project the initial conditions, using the balanced state we just calculated."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "b30935ca",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "theta0.assign(theta_b)\n",
+    "rho0.assign(rho_b)\n",
+    "u0.project(as_vector([10.0, 0.0]))\n",
+    "remove_initial_w(u0)\n",
+    "\n",
+    "state.set_reference_profiles([('rho', rho_b),\n",
+    "                              ('theta', theta_b)])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2afac417",
+   "metadata": {},
+   "source": [
+    "Set up the transport schemes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "7e16a3dc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "supg = True\n",
+    "if supg:\n",
```

Comment:
As in the second book we should probably just pick to use `supg` and get rid of the flag here